homeassistant/components/wolflink/__init__.py

```diff
@@ -41,7 +42,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         gateway_id,
     )
 
-    wolf_client = WolfClient(username, password)
+    wolf_client = WolfClient(
+        username,
+        password,
+        client_lambda=lambda: get_async_client(hass=hass, verify_ssl=False),
```

Comment:
Why do we use a lambda instead of just passing the client instance?

Reply:
I thought there might be multiple clients whose lifecycles are managed and sometimes are dropped. Therefore it seemed to be a good idea to always grab a "fresh" instance. However, as this seems to be unnecessary I replaced it as you suggested.

Reply:
Yeah. If you need to use a different http client you should create a new API client as well, I think.